pkg/apis/lbaas/v1/e2e_helpers_test.go

```diff
@@ -0,0 +1,161 @@
+//go:build integration
+// +build integration
+
+package v1
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"reflect"
+	"time"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var apiClient api.API
+
+type LBaaSE2ETestRun struct {
+	Name string
+	Port int
+}
+
+func waitObject(ctx *context.Context, msg string, o *types.Object, handler func(Gomega, error)) {
+	It(msg, func() {
+		Eventually(func(g Gomega) {
+			err := apiClient.Get(*ctx, *o)
+			handler(g, err)
+		}, 5*time.Minute, 3*time.Second).Should(Succeed())
+	})
+}
+
+func waitObjectReady(ctx *context.Context, o *types.Object) {
+	waitObject(ctx, "eventually is ready", o, func(g Gomega, err error) {
+		Expect(err).NotTo(HaveOccurred())
+
+		hasState, ok := (*o).(StateRetriever)
+		Expect(ok).To(BeTrue())
+
+		Expect(hasState.StateFailure()).To(BeFalse())
```

Comment:
`g` namespaced `Expect` isn't used consistently. not sure if intentional

Reply:
besides that: lgtm

Reply:
it's intentional, but lacks comments about it - will add some

Reply:
adding the comments actually uncovered a wrong usage :upside_down_face: 

Every review is a good review!